airbyte-integrations/connectors/source-klaviyo/source_klaviyo/source.py

```diff
@@ -0,0 +1,99 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+
+from typing import Any, List, Mapping, Tuple, Type
+
+from airbyte_cdk.models import ConnectorSpecification, DestinationSyncMode, SyncMode
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from pydantic import Field
+from pydantic.main import BaseModel
+from source_klaviyo.streams import Campaigns, Events, GlobalExclusions, Lists, Metrics
+
+
+class ConnectorConfig(BaseModel):
+    class Config:
+        title = "Klaviyo Spec"
+
+    api_key: str = Field(
+        description='Klaviyo API Key. See our <a href="https://docs.airbyte.io/integrations/sources/klaviyo">docs</a> if you need help finding this key.',
+        airbyte_secret=True,
+    )
+    start_date: str = Field(
+        description="UTC date and time in the format 2017-01-25T00:00:00Z. Any data before this date will not be replicated.",
+        pattern="^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$",
+        examples=["2017-01-25T00:00:00Z"],
+    )
+
+
+class SourceKlaviyo(AbstractSource):
+    def check_connection(self, logger, config: Mapping[str, Any]) -> Tuple[bool, any]:
+        """Connection check to validate that the user-provided config can be used to connect to the underlying API
+
+        See https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-stripe/source_stripe/source.py#L232
+        for an example.
+
+        :param config:  the user-input config object conforming to the connector's spec.json
+        :param logger:  logger object
+        :return Tuple[bool, any]: (True, None) if the input config can be used to connect to the API successfully, (False, error) otherwise.
+        """
+        ok = False
+        error_msg = None
+        config = ConnectorConfig.parse_obj(config)  # FIXME: this will be not need after we fix CDK
```

Comment:
what do we need to fix?

Reply:
Source should know it's config class and call methods with parsed object and not Dict

Reply:
can you create an issue and add to this comment?